querybook/server/lib/query_analysis/validation/decorators/metadata_decorators.py

```diff
@@ -69,49 +60,41 @@ def _suggest_column_name_if_needed(
                     validation_result.start_ch + len(fuzzy_column_name) - 1
                 )
 
-    def validate(
+    def decorate_validation_results(
         self,
+        validation_results: List[QueryValidationResult],
         query: str,
         uid: int,
         engine_id: int,
-        raw_tokens: List[QueryValidationResult] = None,
         **kwargs,
     ) -> List[QueryValidationResult]:
-        if raw_tokens is None:
-            raw_tokens = self._tokenize_query(query)
-        validation_results = self._validator.validate(
-            query, uid, engine_id, raw_tokens=raw_tokens
-        )
         tables_in_query = self._get_tables_in_query(query, engine_id)
         for result in validation_results:
             self._suggest_column_name_if_needed(result, tables_in_query)
         return validation_results
 
 
-class BaseTableNameSuggester(BaseSQLGlotDecorator):
-    @property
-    def severity(self):
-        return QueryValidationSeverity.WARNING  # Unused, severity is not changed
-
-    @property
-    def message(self):
-        return ""  # Unused, message is not changed
-
+class BaseTableNameSuggester(BaseValidationDecorator):
     @abstractmethod
     def get_full_table_name_from_error(self, validation_result: QueryValidationResult):
         """Returns invalid table name if the validation result is a table name error, otherwise
         returns None"""
         raise NotImplementedError()
 
     def _suggest_table_name_if_needed(
-        self, validation_result: QueryValidationResult
+        self,
+        validation_result: QueryValidationResult,
+        engine_id: int,
     ) -> Optional[str]:
         """Takes validation result and tables in query to update validation result to provide table
         name suggestion"""
         fuzzy_table_name = self.get_full_table_name_from_error(validation_result)
         if not fuzzy_table_name:
             return
-        results, count = search_table.get_table_name_suggestion(fuzzy_table_name)
+        metastore_id = get_query_metastore_id_by_engine_id(engine_id)
+        results, count = search_table.get_table_name_suggestion(
```

Comment:
as the `metastore_id` might be `None`, do we want to skip the suggestion when it is None?